### PR TITLE
feat(workflow_engine): Add type column to detector

### PR DESCRIPTION
**Pull request by wedamija**
We're going to have some way to have different logic for detectors. However we implement it, we'll be using this type column.

### CODE REVIEW

**Review by saponifi3d — APPROVED** [review]
🎉

File: src/sentry/workflow_engine/models/detector.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ class Detector(DefaultFieldsModel, OwnerModel):
         unique=True,
         on_delete=models.SET_NULL,
     )
+    type = models.CharField(max_length=200)
```
Comment (wedamija) [inline src/sentry/workflow_engine/models/detector.py:29]: This is safe to add without a default because this model is completely unused so far.

Comment (saponifi3d) [reply src/sentry/workflow_engine/models/detector.py:29]: 🤞 that ci is okay with this 😅

**Comment by saponifi3d** [comment]
^ CI failure is the `SENTRY_SNAPSHOTS_WRITEBACK=1` one for snapshots, there's also `tests/sentry/backup/test_releases.py` that needs to be executed (so, tests/sentry/backup/test_comparators.py tests/sentry/backup/test_releases.py tests/sentry/backup/test_sanitize.py tests/sentry/tasks/test_relocation.py in all).

**Comment by wedamija** [comment]
> ^ CI failure is the `SENTRY_SNAPSHOTS_WRITEBACK=1` one for snapshots, there's also `tests/sentry/backup/test_releases.py` that needs to be executed (so, tests/sentry/backup/test_comparators.py tests/sentry/backup/test_releases.py tests/sentry/backup/test_sanitize.py tests/sentry/tasks/test_relocation.py in all).

I did run this locally and it didn't generate anything... not sure what I did wrong.

**Review by vgrozdanic — APPROVED** [review]

File: src/sentry/workflow_engine/models/detector.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/workflow_engine/models/detector.py:29]: I disabled checks on the migration so it's fine, but don't follow my example here lol

**Comment by wedamija** [comment]
@vgrozdanic Btw, as part of learning migrations review, in general if you see anyone add `checked=False` to their migration you should block the review. It's generally very unsafe to do so, and unless someone has a lot of experience making database changes in production they might take sentry down by disabling the checks.

**Comment by vgrozdanic** [comment]
> @vgrozdanic Btw, as part of learning migrations review, in general if you see anyone add checked=False to their migration you should block the review. It's generally very unsafe to do so, and unless someone has a lot of experience making database changes in production they might take sentry down by disabling the checks.

Got it, thanks, if it was anyone other but you or Mark, I wouldn't have approved it 😅

**Pull request merged**